### PR TITLE
gzip options tests and simple command test

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/gzip/gzip.go [hunk 1]
```diff
@@ -79,4 +68,30 @@ func main() {
...
+	return nil
+}
+
+func main() {
+	var opts gzip.Options
+	cmdLine.Usage = usage
+
+	if err := opts.ParseArgs(os.Args, cmdLine); err != nil {
+		if errors.Is(err, gzip.ErrStdoutNoForce) {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
```
Comment (rminnich) [inline cmds/core/gzip/gzip.go:81]: can this just  be log.Fatal?

File: cmds/core/gzip/gzip.go [hunk 2]
```diff
@@ -79,4 +68,30 @@ func main() {
...
+			os.Exit(1)
+		}
+		if errors.Is(err, gzip.ErrHelp) {
+			cmdLine.Usage()
+			os.Exit(0)
```
Comment (rminnich) [inline cmds/core/gzip/gzip.go:86]: if it is a usage error, should we exit with 1? This is an error, right?

File: cmds/core/gzip/gzip.go [hunk 3]
```diff
@@ -79,4 +68,30 @@ func main() {
...
+		}
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		cmdLine.Usage()
```
Comment (rminnich) [inline cmds/core/gzip/gzip.go:89]: I wonder if cmdline.Usage might take an io.Writer parameter, and an err parameter? Could all these lines be collapsed into a slightly more capable Usage?

File: cmds/core/gzip/gzip.go [hunk 4]
```diff
@@ -79,4 +68,30 @@ func main() {
...
+		os.Exit(1)
+	}
+
+	if err := run(opts, cmdLine.Args()); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
```
Comment (rminnich) [inline cmds/core/gzip/gzip.go:94]: Could this be a log.Fatal?

File: pkg/gzip/options.go [hunk 5]
```diff
@@ -8,29 +8,33 @@ import (
...
 	"fmt"
-	"os"
 	"path/filepath"
 	"runtime"
 
 	"github.com/klauspost/pgzip"
 )
 
+var (
+	ErrStdoutNoForce = errors.New("standard output is a terminal -- ignoring")
```
Comment (rminnich) [inline pkg/gzip/options.go:18]: Nit: Ignoring what? Also, the text is maybe a bit long? Could it be 
ignoring --force for stdout
of something? I'm just wondering about people searching logs.

File: pkg/gzip/options.go [hunk 6]
```diff
@@ -71,31 +75,31 @@ func (o *Options) ParseArgs(args []string, cmdLine *flag.FlagSet) error {
 		return err
 	}
 
-	return o.validate(len(cmdLine.Args()) > 0)
+	return o.modify(args[0], len(cmdLine.Args()) > 0)
 }
 
-// Validate checks options.
+// modify update options if needed
```
Comment (rminnich) [inline pkg/gzip/options.go:81]: modify update*s*

**Review by rminnich — COMMENTED** [review]
This is a substantial improvement! I had a few questions.

File: pkg/gzip/options.go (hunk 5 above)
Comment (binjip978) [reply pkg/gzip/options.go:18]: "standard output is a terminal -- ignoring" is actually an bsd gzip error, linux have a slightly different error: `gzip: compressed data not written to a terminal. Use -f to force compression.` I will update to something shorter.

File: cmds/core/gzip/gzip.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/gzip/gzip.go:86]: This is intentional, both linux and macOs gzip return 0 if help was requested. Also package logic used empty error to signal that help is requested and it's confusing. So this is not a real error. I'm not a fan of such approach, but we can live with that I think.

**Review by rminnich — APPROVED** [review]

**Pull request merged**